### PR TITLE
adds Harness.hooks_disabled, a context manager

### PR DESCRIPTION
**Pull request by chipaca**
This is a bit of an RFC of an idea that occurred to me while chatting with @axinojolais.

### CODE REVIEW

File: ops/testing.py [hunk 1]
```diff
@@ -15,10 +15,11 @@
 import inspect
 import pathlib
 import random
-from textwrap import dedent
 import tempfile
 import typing
 import yaml
+from contextlib import contextmanager
+from textwrap import dedent
```
Comment (jameinel) [inline ops/testing.py:22]: I'm not clear on pep8 here, are 'from' supposed to be sorted separately (I suppose if you're just 'sort' on the block this is the behavior you get.)

I mostly just want us to be consistent so we don't end up with 2 people editor's fighting over imports. :)

**Review by jameinel — APPROVED** [review]
This looks like a reasonable coding paradigm. I haven't seen any charms trying to disable events, so most of that was hypothetical. If we have concrete examples that end up cleaner with this pattern, I'm happy to have it.

File: ops/testing.py (hunk 1 above)
Comment (dstathis) [reply ops/testing.py:22]: This may be useful

https://github.com/PyCQA/isort

Comment (facundobatista) [reply ops/testing.py:22]: PEP 8 doesn't explicitly say anything about where to put the "from" imports, but examples shown there follow this pattern of grouping all "from" imports at the end of the imports block.

**Comment by chipaca** [comment]
> +1 to this code, if "disabling hooks" is something a user would want to do... is this case a real life need?

the charm in question ended up going a different route but, yes, and the current expectation of doing

    harness.disable_hooks()
    # ... code that needs hooks disabled
    harness.enable_hooks()
    # ... code that needs hooks enabled

is brittle

**Pull request merged**